Update consuming-data.md

bdzekic1 opened the pull request:
There are some formatting and inconsistency issues, so proposing changes. Changes are regarding the `info` and `tip` sections not showing up correctly and some inconsistency in code blocks, but it seems that after changing, they are still not working correctly.

bdzekic1 committed:
Update consuming-data.md

There are some formatting and inconsistency issues, so proposing changes.
Changes are regarding the `info` and `tip` sections not showing up correctly and some inconsistency in code blocks.

diana-macrometa commented on docs/cep/tutorials/consuming-data.md:
```diff
@@ -22,9 +22,9 @@ A source configuration consists of the following:
         Say the stream processor is reading employee records. The employee name might be defined as `emp No` in the source from which you are extracting the records. However, the corresponding attribute name in the stream definition is `employeeNo` because that is how you want to refer to the attribute in the stream processor. In this instance, you need a custom mapping to indicate that `emp No`  ...
-    :::
+:::
 ## Stream Application for Consuming Data
```
Apparently, we used to call stream workers "stream applications." Please update this throughout. "Stream application" should not be in our documentation.

diana-macrometa commented on docs/cep/tutorials/consuming-data.md:
```diff
@@ -79,8 +79,9 @@ Following are the steps to create a stream application:
     
-    !!!info
+:::info
         Mapping is explained in detail in the [Consuming a message in default format](#consume-messages-in-default-format) and [Consuming a message in custom format](#consume-messages-in-custom-format) sections. However, note that you need to add a mapping type to complete a source configuration. If no mapping type i specified, an error is indicated.
```
```suggestion
        Mapping is explained in detail in the [Consuming a message in default format](#consume-messages-in-default-format) and [Consuming a message in custom format](#consume-messages-in-custom-format) sections. However, note that you need to add a mapping type to complete a source configuration. If no mapping type is specified, an error is indicated.
```

diana-macrometa commented on docs/cep/tutorials/consuming-data.md:
```diff
@@ -79,8 +79,9 @@ Following are the steps to create a stream application:
         Mapping is explained in detail in the [Consuming a message in default format](#consume-messages-in-default-format) and [Consuming a message in custom format](#consume-messages-in-custom-format) sections. However, note that you need to add a mapping type to complete a source configuration. If no mapping type i specified, an error is indicated.
+:::
         
 1.  Add stream query to specify how the output is derived and the name of an output stream to which this output is directed.
```
What is a stream query? Are we querying a stream? Or do they mean query worker?

bdzekic1 replied:
A stream query is practically a query inside a stream, for example:
```
insert into PublishSalesTotalsStream
select product, sum(salesValue) as totalSale
from ConsumerSalesTotalsStream WINDOW SLIDING_TIME(1min)
```
and I think we can leave it as it is

diana-macrometa replied:
We do not want to introduce inconsistent terminology. It will confuse the readers (and the writers, lol).

@StoyanTc, is a "stream query" an actual thing? Is there something else that we call them?

StoyanTc replied:
@diana-macrometa - I'm not aware of such term `stream query`. Can't we use just `query a stream`?

diana-macrometa replied:
That would make sense to me. The support gang is helping audit the stream worker content, so we're running across these odd terms in legacy documentation. Probably will be more weird questions as we continue looking through these docs.

@bdzekic1, please change the first part of this step to "Write a query to..."

bdzekic1 replied:
Done

diana-macrometa replied:
@bdzekic1, when you're ready for a re-review, click the circling arrows next to a reviewer's name. GitHub sends them a notification.

bdzekic1 replied:
got it

diana-macrometa commented on docs/cep/tutorials/consuming-data.md:
```diff
@@ -180,8 +183,9 @@ Stream processor consumes a message in the default format when it makes no chang
 
 Stream processor consumes a message in the custom format when it makes changes to the names of the attributes of the message schema before it processes the message. To understand how messages are consumed in custom format, follow the procedure below.
```
What is a "stream processor"? Never heard of that. I know that stream workers sometimes have PROCESS blocks.

bdzekic1 replied:
Stream processor, part of stream worker used for query part, we have sample documentation here: https://macrometa.com/docs/cep/query-guide/functions/rdbms/query , also I would leave it as it is

diana-macrometa replied:
But that is specifically for the RDBMS plugin. Is this topic about using that plugin?

bdzekic1 replied:
_So this should be a proper explanation of stream processors:_
Stream processors are a combination of stream functions and windows. They work directly on the input streams (or named-windows), to generate zero or more new events with zero or more additional output attributes while having the ability to retain and arbitrarily emit events.
They are more advanced than stream functions as they can retain and arbitrarily emit events, and they are more advanced than windows because they can add additional attributes to the events.
_Purpose:_
Stream processors help to achieve complex execution logics that cannot be achieved by other constructs such as functions, aggregate functions, stream functions, and windows.

diana-macrometa replied:
Can you put this someplace logical in this doc? Because that is great content.

bdzekic1 replied:
sure

diana-macrometa commented on docs/cep/tutorials/consuming-data.md:
````diff
@@ -190,11 +194,12 @@ Stream processor consumes a message in the custom format when it makes changes t
         ```
-        :::tip 
+:::tip 
             In the Stream application used as an example in the previous section, assume that when receiving events, the `transNo` attribute is received as `transaction` and the `salesValue` attribute is received as `sales`.  The mapping type is JSON. therefore, you can  add the mappings as JSONPath expressions.
````
Do not update "stream" unless it's in a heading or referring to the UI.

diana-macrometa replied:
```suggestion
            In the stream application used as an example in the previous section, assume that when receiving events, the `transNo` attribute is received as `transaction` and the `salesValue` attribute is received as `sales`. The mapping type is JSON, therefore, you can add the mappings as JSONPath expressions.
```

bdzekic1 committed:
Update consuming-data.md

Changed all the information that had to be changed regarding code samples and documentation. Stream application = stream worker, while stream processor is part of stream worker that's in charge of stream query: https://macrometa.com/docs/cep/query-guide/functions/rdbms/query

diana-macrometa committed:
Merge branch 'master' into patch-9

bdzekic1 committed:
Update consuming-data.md

stream query - query a stream update

bdzekic1 committed:
Update consuming-data.md

I think this would be right place, and FYI I'm not sure if it's on my side, but there seems to be some issue with formatting `note` and `tip` since instead of showing them as blocks, it's stays as plain text `:::tip`

bdzekic1 commented:
Are there any changes left that should be addressed? I think I've covered them all.

diana-macrometa committed:
Merge branch 'master' into patch-9
